test/end-to-end/stock/stock.aggregate_consumption.js

```diff
@@ -0,0 +1,147 @@
+    const getMovementDate = moment(new Date(), 'YYYY-MM-DD').subtract(60, 'days');
+    const getMovementMonth = moment(getMovementDate).month();
+    const getMovementYear = moment(getMovementDate).year();
+
+    const month = ['Jan', 'Fév', 'Mars', 'Avr', 'Mai', 'Juin', 'Juill', 'Août', 'Sept', 'Oct', 'Nov', 'Déc'];
+
+    const fiscalYearLabel = `Fiscal Year ${getMovementYear}`;
+
+    await page.setFiscalPeriod(fiscalYearLabel, month[getMovementMonth]);
+    await page.setDescription(`Aggregate consumption from current depot ${DEPOT_TERTIAIRE}`);
+
+    await page.setHeaderValue(0, 7, 0);
+    await page.setQuantityConsumed(1, 5, 2000);
+    await page.setQuantityLost(1, 6, 1000);
+
+    await page.setQuantityConsumed(2, 5, 500);
+    await page.setQuantityLost(2, 6, 250);
+
+    await page.setHeaderValue(3, 7, 20);
+    await page.setQuantityConsumed(4, 5, 1500);
+    await page.setQuantityLost(4, 6, 3000);
+
+    await page.submitErrorQuantity();
+  });
+
+  it(`Should select the ${DEPOT_TERTIAIRE}`, async () => {
+    await page.changeDepot(DEPOT_TERTIAIRE);
+  });
+
+  it(`Create a new stock aggregate consumption on current depot ${DEPOT_TERTIAIRE}`, async () => {
+    const getMovementDate = moment(new Date(), 'YYYY-MM-DD').subtract(80, 'days');
+    const getMovementMonth = moment(getMovementDate).month();
+    const getMovementYear = moment(getMovementDate).year();
+
+    const month = ['Jan', 'Fév', 'Mars', 'Avr', 'Mai', 'Juin', 'Juill', 'Août', 'Sept', 'Oct', 'Nov', 'Déc'];
+
+    const fiscalYearLabel = `Fiscal Year ${getMovementYear}`;
+
+    await page.setFiscalPeriod(fiscalYearLabel, month[getMovementMonth]);
+    await page.setDescription(`Aggregate consumption from current depot ${DEPOT_TERTIAIRE}`);
+
+    await page.setHeaderValue(0, 7, 0);
+    await page.setQuantityConsumed(1, 5, 250);
+    await page.setQuantityLost(1, 6, 0);
+
+    await page.setQuantityConsumed(2, 5, 250);
+    await page.setQuantityLost(2, 6, 0);
+
+    await page.setHeaderValue(3, 7, 0);
+    await page.setQuantityConsumed(4, 5, 150);
+    await page.setQuantityLost(4, 6, 250);
+
+    await page.submit();
+  });
+
+  it(`Create a complexe aggregate consumption on current depot ${DEPOT_TERTIAIRE}`, async () => {
+    const getMovementDate = moment(new Date(), 'YYYY-MM-DD').subtract(60, 'days');
+    const getMovementMonth = moment(getMovementDate).month();
+    const getMovementYear = moment(getMovementDate).year();
+
+    const getLastDays = new Date(getMovementYear, getMovementMonth + 1, 0);
+
+    const month = ['Jan', 'Fév', 'Mars', 'Avr', 'Mai', 'Juin', 'Juill', 'Août', 'Sept', 'Oct', 'Nov', 'Déc'];
+
+    const fiscalYearLabel = `Fiscal Year ${getMovementYear}`;
+
+    await page.setFiscalPeriod(fiscalYearLabel, month[getMovementMonth]);
+    await page.setDescription(`Aggregate consumption from current depot ${DEPOT_TERTIAIRE}`);
+
+    await page.setHeaderValue(0, 7, 0);
```

Comment:
In general, you should try not to refer to the index in the grid in your tests.  Instead, refer to the name of the row.  Something like
```js
await page.setHeaderValue("Quinine", 7);
```
or even better:
```js
const quinine = page.getInventoryGroup('Quinine');
quinine.setDaysOfStockOut(7);
quinine.setLotQuantityLost("Lot X", 5);
quinine.setLotQuantityConsumed("Lot X", 7);
```

This is much closer to what the user is actually doing.  It also means that other programmers can add lots/inventory items without needing to rewrite these tests.

Reply:
By the way, if you need help with this, I can assist.  I know my descriptions aren't always the best 👍